src/OpenSSL/crypto.py

```diff
@@ -1674,6 +1675,39 @@ def set_time(self, vfy_time):
         _lib.X509_VERIFY_PARAM_set_time(param, int(vfy_time.strftime("%s")))
         _openssl_assert(_lib.X509_STORE_set1_param(self._store, param) != 0)
 
+    def load_locations(self, cafile, capath=None):
+        """
+        Let X509Store know where we can find trusted certificates for the
+        certificate chain.  Note that the certificates have to be in PEM
+        format.
+
+        If capath is passed, it must be a directory prepared using the
+        ``c_rehash`` tool included with OpenSSL.  Either, but not both, of
+        *cafile* or *capath* may be ``None``.
+
+        :param cafile: In which file we can find the certificates (``bytes`` or
+                       ``unicode``).
+        :param capath: In which directory we can find the certificates
+                       (``bytes`` or ``unicode``).
+
+        :return: ``None`` if the locations were set successfully.
+        """
```

Comment:
What happens if both are NULL or both are provided?

Reply:
When both are None, the `X509_STORE_load_locations()` call will return 0, so an `OpenSSL.crypto.Error` will be raised. It's not very helpful, a `ValueError` might have been better, by checking the args explicitly. I did not want to diverge from `OpenSSL.SSL.Context.load_verify_locations()`, which behaves similarly, so I kept it this way. But since `OpenSSL.SSL.Error` is different from `OpenSSL.crypto.Error` anyway, I could add the check and raise a ValueError.

When both are provided, both "locations" will be used. Even better, `load_locations` can be called multiple times, to add more and more locations. Internally, OpenSSL just adds them to the store as `X509_LOOKUP`s. Should I update the docstring with this info?

Reply:
Since pyOpenSSL is generally a thin wrapper around OpenSSL's API I'm fine with it raising `crypto.Error`. If you could update the docstring here then I think this is pretty much good to go. Thanks!

Reply:
I've updated and also extended the docstring to mention that this method can be called multiple times, and added some tests.

One more question: I see that many methods have a `::versionadded` line. Should I add one for `load_locations()`?

Reply:
(... and removed 2 empty lines in another force-push, sorry...)

Reply:
Yes a `.. versionadded:: 20.0` would be appropriate.

Reply:
Done.